src/input_handler.cpp

```diff
@@ -0,0 +1,52 @@
+#include "input_handler.h"
+
+namespace arch {
+    InputHandler InputHandler::input_handler_instance;
+    std::unordered_map<int, std::function<void()>> InputHandler::binded_table;
+    GLFWwindow *InputHandler::_window {};
+    
+    InputHandler::InputHandler() {}
+
+    InputHandler& InputHandler::get() {
```

Comment:
You can initialize `input_handler_instance` to nullptr and then, in this function
```cp
if (input_handler_instance == nullptr)
    input_handler_instance = new ...
return input_handler_instance
```